### PR TITLE
Cable Layers

### PR DESCRIPTION
**Pull request by actioninja**
## About The Pull Request
image

Cables can now be placed onto one of three layers. Layers do not connect, and can have separate powernets running along the same tile.
Machines and terminals only connect to layer 2, yellow.
Cable bridges are made by crafting with the cable coil inhand. They link cables across layers on the same tile, or allow machines to connect to layers other than 2 when placed under them.

## Why It's Good For The Game

I dunno people wanted it

## Changelog
:cl:
add: Cable layers, ctrl click on a cable coil to switch the layer you are placing on
add: Cable bridges can be used to connect layers or connect machines to layers other than yellow
/:cl:

### CODE REVIEW

**Comment by Tharcoonvagh** [comment]
:thinking:

**Comment by Grazyn** [comment]
Shouldn't the knot for layer 1-3 be on the black line of the tile?

**Comment by actioninja** [comment]
> Shouldn't the knot for layer 1-3 be on the black line of the tile?

No.

**Comment by Learner10** [comment]
This is a wonderful addition: perfect for those times when you've got limited space and a heavy need for compact wiring without completely overhauling the wired system and the rooms they're in (TEG in White Ship anyone?), or for separate power nets.

By the by, how will SMES's work if their power terminals are hooked up with multiple cable layers? Like say you're outputting 100 kw? Will the power be evenly split?

**Comment by actioninja** [comment]
I considered handling that and decided it was too much trouble. Machines only link on layer 2, or alternatively you can put a bridge and they'll link to every layer underneath them. All layers will share the same powernet in that case.

**Comment by Learner10** [comment]
Shame. It makes it harder to create an isolated powernet for powering the engine if the wiring layers are all connected that way.

**Comment by optimumtact** [comment]
if someone wants to make a pr to add a way to toggle what layer a machine links on that's fine too

**Comment by 81Denton** [comment]
@actioninja Can you add an examine message that tells people to switch layers with ctrl click? Else this will just end up being forgotten.

File: code/modules/power/cable.dm [hunk 1]
```diff
@@ -382,16 +406,18 @@ GLOBAL_LIST_INIT(cable_coil_recipes, list (new/datum/stack_recipe("cable restrai
...
+/obj/item/stack/cable_coil/Initialize(mapload, new_amount = null)
+	. = ..()
+	pixel_x = rand(-2,2)
+	pixel_y = rand(-2,2)
 	update_icon()
+	recipes = GLOB.cable_coil_recipes
+
+/obj/item/stack/cable_coil/examine(mob/user)
+	. = ..()
+	. += "<b>Ctrl+Click</b> to change the layer you are placing on."
```
Comment (actioninja) [inline code/modules/power/cable.dm]: already done.

Comment (81Denton) [reply code/modules/power/cable.dm]: i am not good with reading

**Comment by optimumtact** [comment]
next step, remove wireless power from APC's and require all machines have cable connection

:imp:

**Comment by Learner10** [comment]
When you say wireless power, are you talking about how some areas outside of the station just have unnatural power, or are you asking for all machines to be wired up? Because that sounds like that would get extremely frustrating for anyone who moves or builds a machine... Engineers would be over-saturated with tedium, all getting overworked, between their normal duties and having to wire up the chem dispenser for Hydroponics, or lay cable for the bartender's new blender... not to mention for the people asking for this stuff would feel having to wait, and all of the mapping changes that would have to be set up EVERYWHERE...

**Comment by optimumtact** [comment]
scared?

**Comment by PKPenguin321** [comment]
Rimworld-esque short range cabling would be neat

**Pull request merged**